x/ibc-transfer/client/cli/tx.go

```diff
@@ -40,24 +37,27 @@ func GetTransferTxCmd(cdc *codec.Codec) *cobra.Command {
 			sender := cliCtx.GetFromAddress()
 			srcPort := args[0]
 			srcChannel := args[1]
-			destHeight, err := strconv.Atoi(args[2])
-			if err != nil {
-				return err
-			}
+			receiver := args[2]
 
-			// parse coin trying to be sent
-			coins, err := sdk.ParseCoins(args[4])
+			coins, err := sdk.ParseCoins(args[3])
 			if err != nil {
 				return err
 			}
 
-			msg := types.NewMsgTransfer(srcPort, srcChannel, uint64(destHeight), coins, sender, args[3])
+			timeoutHeight := viper.GetUint64(flagTimeoutHeight)
+			timeoutTimestamp := viper.GetUint64(flagTimeoutHeight)
```

Comment:
How do we want to use these flags? There are two options:
* They have absolute values (user enters in the exact timeout height and timestamp)
* They have relative values (user enters in timeout height and timestamp relative to the latest retrieved consensus header)

Most users will use the latter, occasionally some user may want the former. The former requires no query to the chain for the latest header. The latter can be achieved by querying for the latest state [`cliCtx.GetNode().Status().SyncInfo.LatestBlockHeight (or BlockTime)`](https://github.com/tendermint/tendermint/blob/1b50b4e7654a6d34d79dbceb1a7c46812c054708/rpc/core/status.go#L61)

Maybe we add a third flag `--relative-timeouts` which is a bool indicating if the timeouts should be relative or absolute (default true). Then enforce that if cliCtx.GenerateOnly is `true` `--relative-timeouts` must be set to false

Reply:
> Most users will use the latter, occasionally some user may want the former. The former requires no query to the chain for the latest header. The latter can be achieved by querying for the latest state `cliCtx.GetNode().Status().SyncInfo.LatestBlockHeight (or BlockTime)`

That's the self block height, you would need to query the `ClientState.LatestHeight()`. See the https://github.com/cosmos/cosmos-sdk/pull/6248#discussion_r428366480 above to use relative timeouts.

Reply:
I'd like additional inputs from @jackzampolin and @zmanian if possible

Reply:
Ah I see. I think my vote would be for still allowing support for both relative and absolute timeouts. Relative timeouts would use the queries described in the linked comments. I think it'd be useful to have a func that gets the latest info of a client state given the port and channel id's. I don't think the overhead is much of a concern since this is just a cli cmd

Reply:
See https://github.com/cosmos/cosmos-sdk/pull/6248#discussion_r428852982

Also, I agree, the overhead is fine for a CLI command. But, I think reducing it by adding a querier that allows grabbing a client associated with a channel is a broadly useful utility for explorers and others.

Reply:
If it's easiest, we could do absolute timeouts for now and add relative timeouts for a follow-up PR.